AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift

```diff
@@ -1022,29 +1022,81 @@ public class ConstructorIO: CIOSessionManagerDelegate {
         ) != nil
     }
 
-    public func obfuscatePIIRequest(request: URLRequest) -> URLRequest {
-        var requestString = request.url!.absoluteString
-        let paths = request.url?.path.components(separatedBy: "/")
-        let params = request.url?.query?.components(separatedBy: "&").map {
-            if $0.contains("=") {
-                $0.components(separatedBy: "=")[1].removingPercentEncoding
-            } else {
-                $0.removingPercentEncoding
+    private func getQueryParams(request: URLRequest) -> [String: String]? {
+        guard let url = request.url, let components = URLComponents(url: url, resolvingAgainstBaseURL: false), let queryItems = components.queryItems else {
+            return nil
+        }
+
+        var keyValuePairs = [String: String]()
+        for item in queryItems {
+            if let name = item.name.removingPercentEncoding, let value = item.value?.removingPercentEncoding {
+                keyValuePairs[name] = value
+            }
+        }
+
+        return keyValuePairs
+    }
+
+    private func replacePathParam(request: URLRequest, name: String, newValue: String) -> URLRequest {
+        guard var components = URLComponents(url: request.url!, resolvingAgainstBaseURL: false) else {
+            return request // Return if URL components cannot be created
+        }
+
+        var newRequest = request
+
+        // Check if the path contains the parameter placeholder
+        var path = components.path
+        if path.contains(name) {
+            path = path.replacingOccurrences(of: name, with: newValue)
+            components.path = path
+            newRequest.url = components.url
+        }
+
+        return newRequest
+    }
+
+    private func replaceQueryParamValue(request: URLRequest, name: String, newValue: String) -> URLRequest {
+        guard var components = URLComponents(url: request.url!, resolvingAgainstBaseURL: false) else {
+            return request // Return if URL components cannot be created
+        }
+
+        var newRequest = request
+
+        if var queryItems = components.queryItems {
+            // Find the query parameter by name and replace its value
+            for (index, item) in queryItems.enumerated() where item.name == name {
+                queryItems[index].value = newValue
+                components.queryItems = queryItems
+                newRequest.url = components.url
+                return newRequest
             }
         }
 
+        // Add the query parameter if it doesn't exist
+        if components.queryItems == nil {
```

Comment:
What cases will this happen? or is this just in case?

Reply:
It's just in case 🤷‍♂️